src/cordova/apple/OutlineAppleLib/Sources/OutlineTunnelSources/OutlineTunnel.swift

```diff
@@ -13,54 +13,156 @@
 
 // Serializable class to wrap a tunnel's configuration.
 // Properties must be kept in sync with ServerConfig in www/types/outlinePlugin.d.ts
 // Note that this class and its non-private properties must be public in order to be visible to the ObjC
 // target of the OutlineAppleLib Swift Package.
 @objcMembers
 public class OutlineTunnel: NSObject, Codable {
-  public var id: String?
-  public var host: String?
-  public var port: String?
-  public var method: String?
-  public var password: String?
-  public var prefix: Data?
-  public var config: [String: String] {
-    let scalars = prefix?.map{Unicode.Scalar($0)}
-    let characters = scalars?.map{Character($0)}
-    let prefixStr = String(characters ?? [])
-    return ["host": host ?? "", "port": port ?? "", "password": password ?? "",
-            "method": method ?? "", "prefix": prefixStr]
-  }
+    public var id: String?
+    public var host: String?
+    public var port: String?
+    public var method: String?
+    public var password: String?
+    public var prefix: Data?
+    public var config: [String: String] {
+        let scalars = prefix?.map{Unicode.Scalar($0)}
+        let characters = scalars?.map{Character($0)}
+        let prefixStr = String(characters ?? [])
+        return ["host": host ?? "", "port": port ?? "", "password": password ?? "",
+                "method": method ?? "", "prefix": prefixStr]
+    }
+    
+    @objc
+    public enum TunnelStatus: Int {
+        case connected = 0
+        case disconnected = 1
+        case reconnecting = 2
+    }
+    
+    public convenience init(id: String, config: [String: Any]) {
+        self.init()
+        self.id = id
+        self.host = config["host"] as? String
+        self.password = config["password"] as? String
+        self.method = config["method"] as? String
+        if let port = config["port"] {
+            self.port = String(describing: port)  // Handle numeric values
+        }
+        if let prefix = config["prefix"] as? String {
+            self.prefix = Data(prefix.utf16.map{UInt8($0)})
+        }
+    }
+    
+    public func encode() -> Data? {
+        return try? JSONEncoder().encode(self)
+    }
+    
+    public static func decode(_ jsonData: Data) -> OutlineTunnel? {
+        return try? JSONDecoder().decode(OutlineTunnel.self, from: jsonData)
+    }
+
+    // Helper function that we can call from Objective-C.
+    @objc public static func getTunnelNetworkSettings(tunnelRemoteAddress: String) -> NEPacketTunnelNetworkSettings {
+        // The remote address is not used for routing, but for display in Settings > VPN > Outline.
+        let settings = NEPacketTunnelNetworkSettings(tunnelRemoteAddress: tunnelRemoteAddress)
 
-  @objc
-  public enum TunnelStatus: Int {
-    case connected = 0
-    case disconnected = 1
-    case reconnecting = 2
-  }
+        // Configure VPN address and routing.
+        let vpnAddress = selectVpnAddress(interfaceAddresses: getNetworkInterfaceAddresses())
```

Comment:
The logic is quite ugly. We can do a lot better.
Perhaps we can add something to and utilities library eventually. It would be great to unify the logic with the other platforms.